### ISSUE TITLE
Borg Confused About Stats

### ISSUE THREAD

**Issue by Netbrian** [angband/angband#6076]
If you press CTRL-Z and 2, it says the borg isn't diving because it's dexterity is below 18/60.  However, his actual dexterity is 18/190.  He should be fine at much deeper levels.
 
@agoodman00 
[Cirialf.zip](https://github.com/user-attachments/files/17234231/Cirialf.zip)

**Comment by agoodman00** [comment]
huh... I am getting that from player->stat_cur... which has that at 38 for you, which seems wrong.  Hopefully the borg code isn't changing it somewhere.

**Comment by Netbrian** [comment]
As a side note, the late-game borg seems to really struggle with getting his dexterity drain.  He keeps picking fights with liches, having his stats drained, and then spending all his time diving through shallow levels.  Not sure how to approach this.

**Comment by agoodman00** [comment]
stat drains are hard on the borg.  The recent (4.x) changes to the way stat drains work and got rid of restore potions.  The result is that things need to be re-coded to make stat drain both more and less important to the borg...  more in that probably once you are at clvl 50, sustain stat should be rewarded more and less in that a mildly drained stat probably shouldn't matter so much to a high level borg.  Additionally the borg should probably be stockpiling ways to get his stats restored.  It is a bunch of work and needs a rethink rather than a quick fix.

**Comment by agoodman00** [comment]
okay I think this is working the way it was designed to work but the way it was designed to work is wrong.  
It looks like it tracks stat_cur - which is the current stat before object bonuses are applied.  So it is saying "your dex, before objects, needs to not have been drained more than 2 points'.  Which it had... but that is a pretty stupid thing to check.  Really it needs to check "your dex, as used..." and I am not sure that is stored anywhere for the borg.
I guess what I am saying is, yes, this is a bug but it isn't going to be an easy fix.  I am going to have to go through the code and check all stat use and update it.

**Comment by Netbrian** [comment]
OK, I've been trying understand the Angband code better and I noticed you replaced this

`reward(value(trait, amt ego missiles) * 1000);`

with this --

`reward(value(trait, amt ego missiles) * 100);`

Was the trait miscalculated somehow?

**Comment by agoodman00** [comment]
Good eyes!  You  can logically look at this and think "This isn't stat calculation, why did Adam make this change"... and here is what happened.  After any major change I run a bunch of random tests, from debugging and seeing if the values are what I think they should be to running the borg and making sure the end behavior matches the change I expect.  One of the tests I run whenever a change is complicated or might relate to "power" calculation (and this change, stat calcs, fits both criteria) is to make sure the dynamic power calculation, as released, matches what happens in the compiled code.  So I fire up the game, load up some character (in this case I used yours because it was nice and complicated) and do ^zp, change borg.txt to use_dynamic_calcs=true then ^z$ (reload borg.txt) ^zp and make sure the numbers match.  Well... they didn't so I spent a bunch of time debugging to figure out where the problem was.  After much debugging one thing I noticed, in borg_power.c lines 1630-1635


    /* cursed arrows are "bad" */
    value -= 1000L * borg.trait[BI_AMISSILES_CURSED];

    /* ego arrows are worth a bonus */
    **value += 100L * borg.trait[BI_AMISSILES_SPECIAL];**

Notice the bonus for ego missiles is 100, not 1000.  Just a mistake I made when I first created the dynamic formulas code.  I corrected it here so the values would match up... so a mistake not related to stats that just happen to be caught while fixing this issue.
